include/hyperion/BGEffectHandler.h

```diff
@@ -34,14 +40,18 @@ private slots:
 	{
 		if(type == settings::BGEFFECT)
 		{
-			const QJsonObject& BGEffectConfig = config.object();
+			_isBgEffectConfigured = false;
+			_bgEffectConfig = config;
 
+			const QJsonObject& BGEffectConfig = _bgEffectConfig.object();
 			#define BGCONFIG_ARRAY bgColorConfig.toArray()
 			// clear background priority
 			_hyperion->clear(PriorityMuxer::BG_PRIORITY);
 			// initial background effect/color
 			if (BGEffectConfig["enable"].toBool(true))
```

Comment:
if (_isBgEffectConfigured = BGEffectConfig["enable"].toBool(true))

Reply:
I prefer not to mix assignments and conditions from a readability perspective.